lsy_drone_racing/envs/real_race_env.py

```diff
@@ -418,6 +494,43 @@ def _jit(self):
                 gate_passed(drone_pos, drone_pos, gate_pos, gate_quat, (0.45, 0.45))
             )
 
+    def update_level_config(self, config: ConfigDict) -> ConfigDict:
+        """Update the level config with the real track objects poses.
+
+        Args:
+            config: A ConfigDict storing the original level configuration.
+
+        Returns:
+            A ConfigDict object, with updated starting pose of gates, obstacles and drones
+        """
+        result_config = config.to_dict()
+
+        # Now the track.randomize needs be set to false,
+        # since we are storing the layout from a real-world placement
+        result_config["env"]["track"]["randomize"] = False
+
+        # Overwrite the original field with the actual positions from the env
+        result_config["env"]["track"]["gates"] = [
+            {
+                "pos": (self.gates.pos[i]).tolist(),
+                "rpy": (R.from_quat(self.gates.quat[i]).as_euler("xyz", degrees=False)).tolist(),
+            }
+            for i in range(self.gates.pos.shape[0])
+        ]
+        result_config["env"]["track"]["obstacles"] = [
+            {"pos": (self.obstacles.pos[i]).tolist()} for i in range(self.obstacles.pos.shape[0])
+        ]
+        result_config["env"]["track"]["drones"] = [
+            {
+                "pos": (self.drones.pos[i]).tolist(),
+                "rpy": (R.from_quat(self.drones.quat[i]).as_euler("xyz", degrees=False)).tolist(),
+                "vel": [0.0, 0.0, 0.0],
+                "ang_vel": [0.0, 0.0, 0.0],
+            }
+            for i in range(self.drones.pos.shape[0])
+        ]
+        return ConfigDict(result_config)
```

Comment:
Does this not work? Again, we want as less indentation, brackets etc as possible to improve readability.
```suggestion
        config = config.copy()
        # We store the real-world track layout, so randomization must be disabled
        config.env.track.randomize = False
        # Overwrite the original positions and orientations with the measured ones
        for i in range(self.gates.pos.shape[0]):
            config.env.track.gates.pos[i] = self.gates.pos[i].tolist()
            config.env.track.gates.rpy[i] = R.from_quat(self.gates.quat[i]).as_euler("xyz").tolist()
        for i in range(self.obstacles.pos.shape[0]):
            config.env.track.obstacles.pos[i] = self.obstacles.pos[i].tolist()
        for i in range(self.drones.pos.shape[0]):
            config.env.track.drones.pos[i] = self.drones.pos[i].tolist()
            config.env.track.drones.rpy[i] = (
                R.from_quat(self.drones.quat[i]).as_euler("xyz").tolist()
            )
        return ConfigDict(config)
```